lib/cyrusdb_twoskip.c

```diff
@@ -513,14 +513,17 @@ static int read_header(struct dbengine *db)
 
     crc = ntohl(*((uint32_t *)(BASE(db) + OFFSET_CRC32)));
 
+    db->end = db->header.current_size;
```

Comment:
Does it matter that we're now setting `db->end` above an error return but we don't do that for any other error return in this function?

Reply:
Pretty sure it's fine - if an error happens here we're going to be aborting the whole database open.